ravenpy/models/emulators.py

```diff
@@ -641,6 +663,160 @@ def ost2raven(self, ops):
         return self.params(*out)
 
 
+class SACSMA(Raven):
+
+    # Details about SAC-SMA:
+    # https://wiki.ewater.org.au/display/SD41/Sacramento+Model+-+SRG
+
+    identifier = "sacsma"
+    templates = tuple((Path(__file__).parent / "raven-sacsma").glob("*.rv?"))
+
+    params = namedtuple(
+        "SACSMAParams", ", ".join(["par_x{:02}".format(i) for i in range(1, 22)])
+    )
+
+    @dataclass
+    class HRU(HRU):
+        land_use_class: str = "FOREST"
+        veg_class: str = "FOREST"
+        soil_profile: str = "DEFAULT_P"
+        aquifer_profile: str = "[NONE]"
+        terrain_class: str = "[NONE]"
+
+    def __init__(self, *args, **kwds):
+        super().__init__(*args, **kwds)
+        self.rvp = RVP(
+            params=SACSMA.params(*((None,) * len(SACSMA.params._fields))),
+            land_use_classes=(
+                LU("FOREST", impermeable_frac=0.0, forest_coverage=0.02345),
+            ),
+        )
+        self.rvh = RVH(hrus=(SACSMA.HRU(),))
+        self.rvt = RVT(**{k: nc() for k in std_vars})
+        self.rvi = RVI(
+            evaporation="PET_OUDIN", rain_snow_fraction="RAINSNOW_DATA"
+        )  # would like to use RAINSNOW_HBV??
```

Comment:
Try to put the comments on top of lines instead of at the end, as Black's formatting makes long line a bit weird.

Reply:
Sorry. Should remove this comment entirely. It's a remainder of my testing

Reply:
Done.